src/background_thread.c

```diff
@@ -799,7 +799,13 @@ background_thread_stats_read(tsdn_t *tsdn, background_thread_stats_t *stats) {
 	nstime_init(&stats->run_interval, 0);
 	for (unsigned i = 0; i < max_background_threads; i++) {
 		background_thread_info_t *info = &background_thread_info[i];
-		malloc_mutex_lock(tsdn, &info->mtx);
+		if (malloc_mutex_trylock(tsdn, &info->mtx)) {
+			/*
+			 * Each background thread run may take a long time;
+			 * avoid waiting on the stats if the thread is active.
+			 */
+			continue;
```

Comment:
Do you want to have a counter or something so that the caller can know how many background threads succeeded? E.g. instead of returning a `bool`, return an `unsigned` counter so that the caller can know more details if desired? May not be immediately useful though given the current way this function is called.

Reply:
Yep it could be useful if we decide to report detailed per thread stats -- right now only the aggregated numbers are shown.